plugins/mac-virtualcam/src/obs-plugin/plugin-main.mm

```diff
@@ -370,15 +370,17 @@ static bool virtualcam_output_start(void *data)
                                            .mElement = kCMIOObjectPropertyElementMain};
         CMIOObjectGetPropertyDataSize(kCMIOObjectSystemObject, &address, 0, NULL, &size);
         size_t num_devices = size / sizeof(CMIOObjectID);
-        CMIOObjectID cmio_devices[num_devices];
-        CMIOObjectGetPropertyData(kCMIOObjectSystemObject, &address, 0, NULL, size, &used, &cmio_devices);
+        NSMutableData *cmioDevices = [NSMutableData dataWithLength:size];
```

Comment:
This code doesn't work - the example I posted was more or less pseudocode, this needs to be implemented properly and tested as well (`CMIOObjectGetPropertyData` expects a void pointer, not an `NSMutableData` pointer).

Reply:
I don't a have a Apple machine so I can't test this code.
I reverted to pragmas, feel free to do a PR which use NSMutableData.

Reply:
If desired (I think it would be a better fix than throwing `pragma clang diagnostic push` all over the code), I can look into this.

Reply:
It surely does but I don't have what is needed to correctly test that, so feel free.